osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/TimelinePart.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using OpenTK;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+
+namespace osu.Game.Screens.Edit.Components.Timelines.Summary.Parts
+{
+    /// <summary>
+    /// Represents a part of the summary timeline..
+    /// </summary>
+    internal abstract class TimelinePart : CompositeDrawable
+    {
+        private readonly Container timeline;
+
+        protected TimelinePart()
+        {
+            AddInternal(timeline = new Container { RelativeSizeAxes = Axes.Both });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuGameBase osuGame)
+        {
+            osuGame.Beatmap.ValueChanged += b =>

```

Comment:
Really shouldn't be doing this. the binding will remain even when the editor is closed.

The beatmap should be passed into the timeline and down to components.

If you really want to bind against the game's Beatmap then the SummaryTimeline should be doing this, making its *own* bindable instance (via `GetBindable` etc) and then passing that down to its components.